var/spack/repos/builtin/packages/git/package.py

```diff
@@ -199,6 +203,10 @@ def setup_environment(self, spack_env, run_env):
             spack_env.append_flags('CFLAGS', '-I{0}'.format(
                 self.spec['gettext'].prefix.include))
 
+        # gitk requires tk's wish to be found in $PATH.
+        if '+tcltk' in self.spec:
+            run_env.prepend_path('PATH', self.spec['tk'].prefix.bin)
```

Comment:
I think it's better to check:
```python
if 'tk' in self.spec:
   run_env.prepend_path('PATH', self.spec['tk'].prefix.bin)
```
The way this statement is written right now is bound to fail when git is declared as an external (as the variant might be there, but the dependency is not).

Reply:
@adamjstewart @KineticTheory Forgot to mention: I don't know `tcl` and `tk` at all, but it seems to me that the situation here might be common for packages that depend on `tk`. If somebody could confirm this impression, I think the proper place where to modify the `run_env` is the `setup_dependnent_environment` of the `tk` package.

Reply:
@alalazo I must confess that I too am blissfully ignorant of tcl/tk environments.  :relieved: 

I'll implement your recommended changes in my branch and do a couple of test builds before I push the updates.